### PR TITLE
remove deprecated annotations in server-service template

### PR DESCRIPTION
**Pull request by kschoche**
Changes proposed in this PR:
- We previously relied on setting an annotation ` service.alpha.kubernetes.io/tolerate-unready-endpoints: "true"` which has been deprecated for 5 years and replaced with `spec.publicNotReadyAddresses` which we also already use.
- Remove it and continue to use the correct spec field on the server-service.yaml.

https://github.com/kubernetes/kubernetes/pull/63742
https://github.com/kubernetes/kubernetes/issues/63741

How I've tested this PR:
CI and Unit tests should pass.
Bats test updated.

How I expect reviewers to test this PR:
👀 

Checklist:
- [x] Tests added
- [x] CHANGELOG entry added 
  > HashiCorp engineers only, community PRs should not add a changelog entry.
  > Entries should use present tense (e.g. Add support for...)

### CODE REVIEW

File: charts/consul/test/unit/server-service.bats [hunk 1]
```diff
@@ -103,15 +98,6 @@ load _helpers
...
 # annotations
 
-@test "server/Service: one annotation by default" {
-  cd `chart_dir`
-  local actual=$(helm template \
-      -s templates/server-service.yaml  \
-      . | tee /dev/stderr |
-      yq -r '.metadata.annotations | length' | tee /dev/stderr)
-  [ "${actual}" = "1" ]
-}
```
Comment (ishustava) [inline charts/consul/test/unit/server-service.bats:113]: Could this be changed to `no annotation by default` case?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,9 @@
 ## UNRELEASED
 
+IMPROVEMENTS:
+* Helm:
+  * Remove use of deprecated annotation `service.alpha.kubernetes.io/tolerate-unready-endpoints: "true"` in server-service template. [[GH-1619](https://github.com/hashicorp/consul-k8s/pull/1619)]
```
Comment (ishustava) [inline CHANGELOG.md:5]:
```suggestion
* Helm:
  * Remove deprecated annotation `service.alpha.kubernetes.io/tolerate-unready-endpoints: "true"` in the `server-service` template. [[GH-1619](https://github.com/hashicorp/consul-k8s/pull/1619)]
```

**Review by ishustava — APPROVED** [review]
Looks good! Just a couple of minor comments

**Review by thisisnotashwin — APPROVED** [review]
NICE!

File: charts/consul/test/unit/server-service.bats (hunk 1 above)
Comment (kschoche) [reply charts/consul/test/unit/server-service.bats:113]: Totes!

**Pull request merged**